airbyte-integrations/connectors/source-newsdata/source_newsdata/newsdata.yaml

```diff
@@ -14,13 +14,25 @@ definitions:
   base_retriever:
     record_selector:
       $ref: "*ref(definitions.selector)"
-    paginator:
-      type: NoPagination
   base_stream:
     retriever:
       $ref: "*ref(definitions.base_retriever)"
       requester: 
         $ref: "*ref(definitions.base_requester)"
+  cursor_paginator:
+    type: "DefaultPaginator"
+    pagination_strategy:
+      type: "CursorPagination"
+      cursor_value: "{{ response['nextPage'] }}"
+      page_size: 10
+      # page_size: !!int "{% if config['is_free_tier'] %}{{ 10 }}{% else %}{{ 50 }}{% endif %}"
```

Comment:
The API sets the `page_size` dependent on the subscription you have.

- Free tier gets `page_size = 10`
- Any paid tier gets `page_size = 50`

The jinja interpolation works, but in this case, as it interpolates after !!int is executed, it fails to convert the string to an int. We need to cast it to int, as it is the type of `page_size` property. Any hint?

Reply:
@girarda any advice here?

Reply:
I would suggest setting the page_size in the spec instead of the is_free_tier field. I'll create an issue for the pagination strategy to handle dynamic page_size

Reply:
@girarda, but wouldn't we still need string interpolation if handled in the spec?

Reply:
yes, you're right. There's no way to dynamically set the page size today

Reply:
Okey, so for now I will remove `is_free_tier` from `spec.yaml`, and set the `page_size = 10`, so that this connector works for both, free and paid tiers.

We will leave the dynamic `page_size` for a follow up.